006-tls-encrypt-the-kafka-connect-rest-api.md

```diff
@@ -0,0 +1,110 @@
+
+```
+listeners: https://:8443
+rest.advertised.listener: https
+rest.advertised.port: 8443
+listeners.https.ssl.client.auth: none
+listeners.https.ssl.truststore.location: /tmp/kafka/kafka-connect-rest.truststore.p12
+listeners.https.ssl.truststore.password: ***generated password***
+listeners.https.ssl.truststore.type: PKCS12
+listeners.https.ssl.keystore.location: /tmp/kafka/kafka-connect-rest.keystore.p12
+listeners.https.ssl.keystore.password: ***generated password***
+listeners.https.ssl.keystore.type: PKCS12
+```
+
+To communicate with the encrypted Connect REST API listener, the `KafkaMirrorMaker2`, `KafkaConnect` and `KafkaConnector` operators require modifications to the `AbstractConnectOperator` to apply the TLS certs as truststore options on the Vertx WebClient.
+
+### Enabling a plain HTTP REST API listener
+
+A new field is added to the `KafkaConnect` and `KafkaMirrorMaker2` specs named `restListeners`, which allows users to configure an additional unencrypted HTTP listener on port 8083 in the Kafka Connect configuration.
+This enables users to access the REST API without TLS if required.
+The `KafkaConnector`, `KafkaConnect` and `KafkaMirrorMaker2` operators will continue to use the TLS encrypted listener when an unencrypted HTTP listener is enabled.
+
+The `restListeners` field is designed for potential future extension, and is a list of items that contain a `protocol` field.
+For this proposal, the `protocol` field only supports the value `http`. One (or more) items in the `restListeners` list will add the unencrypted HTTP listener on port 8083.
+
+For example, the following CR will enable the HTTP listener on port 8083:
+```
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaConnect
+spec:
+  restListeners:
+  - protocol: http
+  ...
+```
+
+The schema for `restListeners` is as follows:
+```
+openAPIV3Schema:
+  type: object
+  properties:
+    spec:
+      type: object
+      properties:
+        restListeners:
+          type: array
+          items:
+            type: object
+            properties:
+              protocol:
+                type: string
+                enum:
+                - http
+                description: The protocol of the REST listener.
+            required:
+            - protocol
+          description: List of additional REST listeners.
+        ...
+```
+
+
+### Certificate renewals
+
+The self-signed TLS certificate will have a 1-year expiration.
+Thirty day before the self-signed TLS certificate expires, the operator will automatically renew the certificate, replace the old self-signed certificate, and restart the pods in a controlled manner to ensure the new certificates are in use by Kafka Connect.
+
+This would need to be a multi-phase process consisting of the following steps:
+
+1. Generate new certificate
+2. Distribute the public key to all pods and cluster operator truststores
```

Comment:
Let's use the term certificate rather than public key, since a cert contains more information than just a public key.
```suggestion
2. Distribute the new certificate to all pods and cluster operator truststores
```

Reply:
Updated